libs/plugins/stream-client/src/lib/stream-designer/stream-designer.component.ts

```diff
@@ -1,14 +1,79 @@
-import { Component, OnInit } from '@angular/core';
-import { ActivatedResourceRoute } from '@flogo-web/lib-client/core';
-
+import { Component, HostBinding, OnDestroy, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { LanguageService } from '@flogo-web/lib-client/language';
+import {
+  animateChild,
+  transition,
+  trigger as animationTrigger,
+} from '@angular/animations';
+import { NotificationsService } from '@flogo-web/lib-client/notifications';
+import { select, Store } from '@ngrx/store';
+import {
+  ChangeDescription,
+  ChangeName,
+  FlogoStreamState,
+  selectStreamState,
+  StreamService,
+} from '../core';
+import { StreamStoreState as AppState } from '../core';
+import { takeUntil } from 'rxjs/operators';
+import { SingleEmissionSubject } from '@flogo-web/lib-client/core';
 @Component({
   selector: 'flogo-stream-designer',
   templateUrl: './stream-designer.component.html',
   styleUrls: ['./stream-designer.component.less'],
+  animations: [
+    animationTrigger('initialAnimation', [transition('void => *', animateChild())]),
+  ],
 })
-export class StreamDesignerComponent implements OnInit {
-  // todo: making it public for demo only
-  constructor(public activatedResource: ActivatedResourceRoute) {}
+export class StreamDesignerComponent implements OnInit, OnDestroy {
+  @HostBinding('@initialAnimation') initialAnimation = true;
+
+  public streamState: FlogoStreamState;
+  public isStreamMenuOpen = false;
+
+  private ngOnDestroy$ = SingleEmissionSubject.create();
+
+  constructor(
+    private store: Store<AppState>,
+    private _streamService: StreamService,
```

Comment:
can we remove the leading underscore for the private fields? Just to maintain consistency.